src/compiler/checker.ts

```diff
@@ -29143,6 +29142,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return errorType;
         }
 
+        if (classDeclarationExtendsNull(classLikeDeclaration)) {
+            return isCallExpression ? errorType : nullWideningType;
+        }
+
         const classType = getDeclaredTypeOfSymbol(getSymbolOfDeclaration(classLikeDeclaration)) as InterfaceType;
         const baseClassType = classType && getBaseTypes(classType)[0];

```

Comment:
In a normal situation, there would be no base class in the case of `extends null`. `resolveBaseTypesOfClass` returns an empty array if the `baseConstructorType` there is not an object (while `null` is a valid return value of `getBaseConstructorTypeOfClass`)

We don't have direct access to this resolution though, all that we currently have are those base types here and that's roughly equivalent of:
```ts
type BaseTypeOfClassArr = [ConstructorType?]
type BaseTypesOfInterfaces = [...Type[]]
type BaseTypes = [...BaseTypeOfClassArr, ...BaseTypesOfInterfaces]
```

There is always at most one base type of class but it's also optional. If it's there then it's always the first item in the array and that's why `[0]` is used here. So the problem was caused here by the fact that this array was not empty and yet the first item in the array was not a base type of class. That allowed the code to continue here, returning a value that didn't have any construct signatures (`null`).